src/Catalog/Registration/RegistrationCatalogEntry.cs

```diff
@@ -1,12 +1,15 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using VDS.RDF;
 
 namespace NuGet.Services.Metadata.Catalog.Registration
 {
+    public delegate bool ShouldIncludeRegistrationPackage(RegistrationEntryKey key, string resourceUri, IGraph graph);
```

Comment:
Not sure what you mean by:

> move the delegate inside ShouldIncludeRegistrationPackage

The delegate itself is called `ShouldIncludeRegistrationPackage`. I chose not to put it inside `RegistrationCatalogEntry` because this delegate is passed from a couple levels up, which do not have knowledge of `RegistrationCatalogEntry`.

In general, delegates are just types so I see no reason why they should be nested.